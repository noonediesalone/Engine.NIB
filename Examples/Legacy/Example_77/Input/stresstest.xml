<StressTesting>
  <UseSpreadedTermStructures>true</UseSpreadedTermStructures>
  <StressTest id="parallel_vols">

    <DiscountCurves>
    </DiscountCurves>

    <IndexCurves>
    </IndexCurves>

    <YieldCurves>
    </YieldCurves>

    <FxSpots>
    </FxSpots>

    <FxVolatilities>
      <FxVolatility ccypair="USDEUR">
        <ShiftType>Absolute</ShiftType>
        <Shifts>0.1</Shifts>
        <ShiftExpiries>1Y</ShiftExpiries>
      </FxVolatility>
    </FxVolatilities>


    <SwaptionVolatilities>
    </SwaptionVolatilities>


    <CapFloorVolatilities>
    </CapFloorVolatilities>


    <EquitySpots>
    </EquitySpots>


    <EquityVolatilities>
    </EquityVolatilities>
    <SecuritySpreads />
    <RecoveryRates />
    <SurvivalProbabilities />
  </StressTest>

  <StressTest id="explicit_vols">

    <DiscountCurves>
    </DiscountCurves>

    <IndexCurves>
    </IndexCurves>

    <YieldCurves>
    </YieldCurves>

    <FxSpots>
    </FxSpots>

    <FxVolatilities>
      <FxVolatility ccypair="USDEUR">
        <ShiftType>Absolute</ShiftType>
        <Shifts>0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.3</Shifts>
        <ShiftExpiries>6M, 1Y, 2Y, 4Y, 5Y, 7Y, 10Y</ShiftExpiries>
      </FxVolatility>
    </FxVolatilities>


    <SwaptionVolatilities>
    </SwaptionVolatilities>


    <CapFloorVolatilities>
    </CapFloorVolatilities>


    <EquitySpots>
    </EquitySpots>


    <EquityVolatilities>
    </EquityVolatilities>
    <SecuritySpreads />
    <RecoveryRates />
    <SurvivalProbabilities />
  </StressTest>

  <StressTest id="weighting_off">

    <DiscountCurves>
    </DiscountCurves>

    <IndexCurves>
    </IndexCurves>

    <YieldCurves>
    </YieldCurves>

    <FxSpots>
    </FxSpots>

    <FxVolatilities>
      <FxVolatility ccypair="USDEUR">
        <ShiftType>Absolute</ShiftType>
        <WeightedShifts>
          <WeightingSchema>Unadjusted</WeightingSchema>
          <Shift>0.1</Shift>
          <Tenor>1Y</Tenor>
        </WeightedShifts>
      </FxVolatility>
    </FxVolatilities>


    <SwaptionVolatilities>
    </SwaptionVolatilities>


    <CapFloorVolatilities>
    </CapFloorVolatilities>


    <EquitySpots>
    </EquitySpots>


    <EquityVolatilities>
    </EquityVolatilities>
    <SecuritySpreads />
    <RecoveryRates />
    <SurvivalProbabilities />
  </StressTest>

  <StressTest id="weighting">

    <DiscountCurves>
    </DiscountCurves>

    <IndexCurves>
    </IndexCurves>

    <YieldCurves>
    </YieldCurves>

    <FxSpots>
    </FxSpots>

    <FxVolatilities>
      <FxVolatility ccypair="USDEUR">
        <ShiftType>Absolute</ShiftType>
        <WeightedShifts>
          <WeightingSchema>Weighted</WeightingSchema>
          <Shift>0.1</Shift>
          <Tenor>1Y</Tenor>
          <ShiftWeights>0.5,1.0,2.0,3.0,4.0,5.0,6.0</ShiftWeights>
          <WeightTenors>6M,1Y,2Y,4Y,5Y,7Y,10Y</WeightTenors>
        </WeightedShifts>
      </FxVolatility>
    </FxVolatilities>


    <SwaptionVolatilities>
    </SwaptionVolatilities>


    <CapFloorVolatilities>
    </CapFloorVolatilities>


    <EquitySpots>
    </EquitySpots>


    <EquityVolatilities>
    </EquityVolatilities>
    <SecuritySpreads />
    <RecoveryRates />
    <SurvivalProbabilities />
  </StressTest>
</StressTesting>