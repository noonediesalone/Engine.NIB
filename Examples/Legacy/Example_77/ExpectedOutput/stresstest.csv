#TradeId,ScenarioLabel,Base NPV,Scenario NPV,Sensitivity
FX_CALL_OPTION_1Y,explicit_vols,75059.13,113030.18,37971.05
FX_CALL_OPTION_1Y,parallel_vols,75059.13,113030.18,37971.05
FX_CALL_OPTION_1Y,weighting,75059.13,113030.18,37971.05
FX_CALL_OPTION_1Y,weighting_off,75059.13,113030.18,37971.05
FX_CALL_OPTION_2Y,explicit_vols,107489.87,187137.58,79647.71
FX_CALL_OPTION_2Y,parallel_vols,107489.87,160435.33,52945.45
FX_CALL_OPTION_2Y,weighting,107489.87,213770.07,106280.20
FX_CALL_OPTION_3Y,explicit_vols,137141.19,249028.69,111887.49
FX_CALL_OPTION_3Y,parallel_vols,137141.19,200953.32,63812.13
FX_CALL_OPTION_3Y,weighting,137141.19,296548.37,159407.18
FX_CALL_OPTION_5Y,explicit_vols,194124.90,390308.63,196183.73
FX_CALL_OPTION_5Y,parallel_vols,194124.90,273370.16,79245.26
FX_CALL_OPTION_5Y,weighting,194124.90,499707.58,305582.67
